torch/nn/parallel/distributed.py

```diff
@@ -372,6 +377,13 @@ def model_parameters(m):
 
             for module_copy in self._module_copies[1:]:
                 for param, copy_param in zip(self.module.parameters(), parameters(module_copy)):
+                    # Reducer requires param copies have the same strides across replicas.
+                    # Fixes up copy_param strides in case replicate didn't match param strides.
+                    if param.layout is torch.strided and param.stride() != copy_param.stride():
```

Comment:
Do we expect this to occur with current `replicate.py` implementation?

Reply:
Yes, replicate transposes to rowmajor contiguous as it broadcasts.  Allowing format-preserving intra-process broadcasts is nice to have but out of scope.  Code here intends to make DDP agnostic to implementation details of replicate, so we're free to change replicate (intra-process broadcast) without breaking DDP.  This only happens once on construction, so a few deep copies aren't terrible.